### PR TITLE
Packages: Move design-system to new structure 

### LINKED ISSUE #6051: Packages: Move `design-system` to new structure
## Feature Description

* [ ] Move code from `/assets/src/design-system` to `/packages/design-system/src/`
* [ ] Update imports where necessary.  
  Use `@web-stories-wp/...` imports for anything that is already a package, relative imports for everything else.
* [ ] Create `/packages/design-system/package.json` referencing its dependencies
* [ ] Create `/packages/design-system/README.md` with short documentation.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Move design-system to new package structure. 

## Relevant Technical Choices

All references to import of `useContext` from use-context-selector now use the design package. 

Worth noting, there are still 3 places that use reacts version of `useContext`. I tried changing these as well but it made test fail. Maybe something to  look at later. 
Update the package files to remove packages only used in design package. 
Here I fixed a JS unit test that made this fail. https://github.com/google/web-stories-wp/pull/8087/commits/aa91c0a2721ca500462f196fd06fcaa511932f87 It feels like it was unrelated, but it was blocking this PR. 

## To-do

#7875 should be fixed next.

## User-facing changes

N/A

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

N/A

### Does this PR change what data or activity we track or use?

N/A

### Does this PR have a legal-related impact?

N/A

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6051

### CODE REVIEW

File: assets/src/edit-story/app/highlights/quickActions/useQuickActions.js
Comment (swissspidy) [inline assets/src/edit-story/app/highlights/quickActions/useQuickActions.js]:
```suggestion
/** @typedef {import('@web-stories-wp/design-system').MenuItemProps} MenuItemProps */
```

File: packages/design-system/src/components/mediaInput/index.js
Comment (swissspidy) [inline packages/design-system/src/components/mediaInput/index.js:42]: Why is this re-exported here? Shouldn't other code just import from `./constants` as well?

File: assets/src/edit-story/components/helpCenter/toggle/test/index.js
Comment (swissspidy) [inline assets/src/edit-story/components/helpCenter/toggle/test/index.js:20]: Note to self: good reminder to check how much sense it makes to share test utils like this

File: packages/design-system/src/components/mediaInput/index.js
Comment (spacedmonkey) [reply packages/design-system/src/components/mediaInput/index.js:42]: > Why is this re-exported here

I was not exported before, but directly access. This const, needs to be available globally. It can then be exposed in 
`export { MediaInput, MEDIA_VARIANTS } from './mediaInput';` in components/index.js

File: assets/src/edit-story/components/helpCenter/toggle/test/index.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/helpCenter/toggle/test/index.js:20]: There was no way around this. Including this in the root, broke many other tests.

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (swissspidy) [inline assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: What's the reason for the change to use `forwardRef` in this component here?

**Comment by spacedmonkey** [comment]
-26 kB of savings 🎉

File: packages/design-system/src/components/hex/index.js
Comment (swissspidy) [inline packages/design-system/src/components/hex/index.js]:
```suggestion
```

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]:
```
  console.error
    Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

    Check the render method of `FontPicker`.
        at mockConstructor (/Users/jonny/workspace/web-stories-wp/node_modules/jest-mock/build/index.js:171:19)
        at FontPicker (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/fontPicker.js:38:3)
        at div
        at row__Row (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at FontControls (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/font.js:64:3)
        at div
        at content__Container (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Content (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/shared/content.js:40:20)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at section
        at panel__Wrapper (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Panel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/panel.js:45:3)
        at SimplePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/simplePanel.js:29:24)
        at textStyle___StyledSimplePanel (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at StylePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/textStyle.js:34:27)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at Wrapper (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/test/textStyle.js:57:20)
        at TestPanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/shared/test/_utils.js:34:3)
        at Object.<anonymous>.exports.ThemeProvider (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/ServerStyleSheet.js:66:13)

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:67:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:43:5)
      at validateFunctionComponentInDev (node_modules/react-dom/cjs/react-dom.development.js:17926:9)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17893:7)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:19049:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23940:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22779:12)

  console.error
    Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

    Check the render method of `ColorControls`.
        at mockConstructor (/Users/jonny/workspace/web-stories-wp/node_modules/jest-mock/build/index.js:171:19)
        at div
        at row__Row (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at ColorControls (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/color.js:29:26)
        at div
        at content__Container (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Content (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/shared/content.js:40:20)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at section
        at panel__Wrapper (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Panel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/panel.js:45:3)
        at SimplePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/simplePanel.js:29:24)
        at textStyle___StyledSimplePanel (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at StylePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/textStyle.js:34:27)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at Wrapper (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/test/textStyle.js:57:20)
        at TestPanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/shared/test/_utils.js:34:3)
        at Object.<anonymous>.exports.ThemeProvider (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/ServerStyleSheet.js:66:13)

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:67:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:43:5)
      at validateFunctionComponentInDev (node_modules/react-dom/cjs/react-dom.development.js:17926:9)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17893:7)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:19049:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23940:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22779:12)
```

I got the following javascript unit test error. I believe it is unrelated, but it did  block this PR. It is  worth note, this warning was in other test runs, but for some reason it change to a error here.

Comment (swissspidy) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: There's a dedicated ticket for this: #7875

cc @miina

File: assets/src/edit-story/components/helpCenter/toggle/test/index.js
Comment (swissspidy) [reply assets/src/edit-story/components/helpCenter/toggle/test/index.js:20]: Yes I get that. Still something I'd like to look into as it might not be the best way to share test utils like this.

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: I believe that @miina is away ATM. 

The difference is, some reason, it become an error over a warning here. @swissspidy should I skip the test for now?

Comment (swissspidy) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: Let's fix #7875 first in a separate PR. It might be fixed by the change you did here, but I'd want this to be in a dedicated PR & reviewed by @miina.

File: assets/src/edit-story/components/helpCenter/toggle/test/index.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/helpCenter/toggle/test/index.js:20]: I agree. Should this be in a different package? 

I would like to handle this in a different PR, this PR is massive as it is!

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: Broken the fix into it's own PR - https://github.com/google/web-stories-wp/pull/8100. 

I have skipped the test for now.

File: assets/src/dashboard/app/views/shared/savedTemplateGridView.js
Comment (miina) [inline assets/src/dashboard/app/views/shared/savedTemplateGridView.js:25]: Is there a reason to remove the extra line between the import and the starting comment? Looks like it's done in various places in the files. Isn't it more readable when there's an empty line between the import and the next comment?

File: docs/design-system.md
Comment (miina) [inline docs/design-system.md]: Here there's still `assets/src/design-system/theme` in one place.

**Comment by miina** [comment]
Note: There is still a reference to `assets/src/design-system` in the `.eslintrc` file and also in the `.storybook/main.cjs` file.

File: assets/src/dashboard/app/views/shared/savedTemplateGridView.js
Comment (miina) [reply assets/src/dashboard/app/views/shared/savedTemplateGridView.js:25]: @spacedmonkey This comment was marked resolved but I can still see various places where there is no empty line between the import and the comment block. Thoughts?

(Marked it unresolved until then)

Comment (spacedmonkey) [reply assets/src/dashboard/app/views/shared/savedTemplateGridView.js:25]: Fixed now.

File: assets/src/edit-story/components/panels/document/publish/publishTime.js
Comment (swissspidy) [inline assets/src/edit-story/components/panels/document/publish/publishTime.js]: Does using `DropDown` instead of `DropDownSelect` (which is a child of `DropDown`) cause any issues here?

cc @miina @BrittanyIRL just in case

Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/document/publish/publishTime.js]: i believe this is because prometheus wanted to do something else with the exposed menu, and so just the select was used. haven't looked super close.

File: assets/src/edit-story/components/form/select/select.js
Comment (BrittanyIRL) [inline assets/src/edit-story/components/form/select/select.js]: @miina did you see this one, is this ok?

Comment (swissspidy) [reply assets/src/edit-story/components/form/select/select.js]: Why is `DropDown` imported as `DropDownSelect` here? Why the renaming? 🤔

**Review by BrittanyIRL — COMMENTED** [review]
`edit-story/components/checklist/styles` got merged this am and uses `Text` and `THEME_CONSTANTS` which is the only linting error getting caught here.  Read through the whole PR. Running locally through things visually now, will approve after that.

**Review by BrittanyIRL — APPROVED** [review]
LGTM, this is a big lift. Made the fix locally to that one file that got merged to main that's got the old paths and everything is passing and working great.

File: assets/src/edit-story/components/panels/document/publish/publishTime.js
Comment (miina) [reply assets/src/edit-story/components/panels/document/publish/publishTime.js]: It doesn't seem to cause any issues here, however, we indeed needed only the select button here for opening the calendar.
 @spacedmonkey Why use `DropDown` instead of `DropDownSelect` here?

File: assets/src/edit-story/components/form/select/select.js
Comment (miina) [reply assets/src/edit-story/components/form/select/select.js]: Thanks for tagging. It doesn't make a difference since it's just renaming it but not sure why -- @spacedmonkey Any specific reason for renaming `DropDown` to `DropDownSelect` here? Let's change it back, if not.

File: assets/src/edit-story/components/panels/document/publish/publishTime.js
Comment (swissspidy) [reply assets/src/edit-story/components/panels/document/publish/publishTime.js]: Looks like DropDownSelect is not exported from the design system package.

Comment (miina) [reply assets/src/edit-story/components/panels/document/publish/publishTime.js]: Yep, perhaps it should be exported. It seems to work as-is, too, since by default the dropdown is displaying only the select button anyway, however, there's some other logic that's not needed, ideally we'd keep using the select only for avoiding any potential unexpected behaviour.

Would need some investigation before just switching to using `DropDown`.

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (swissspidy) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: I'd rather want to fix the actual bug instead of just skipping tests and hiding potential problems.

Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: > I'd rather want to fix the actual bug instead of just skipping tests and hiding potential problems

Can we not fix it in #7875 ? I personally tried to fix it and I don't know how.

File: assets/src/edit-story/components/form/select/select.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/form/select/select.js]: Fixed in https://github.com/google/web-stories-wp/pull/8087/commits/eecb244e0bc64ee01035f6d538a6e57c6e0cc59a

File: assets/src/edit-story/components/panels/document/publish/publishTime.js
Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/document/publish/publishTime.js]: Fixed in https://github.com/google/web-stories-wp/pull/8087/commits/eecb244e0bc64ee01035f6d538a6e57c6e0cc59a

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: i put this on our plate for the next sprint, we just are swamped this week trying to finish our work for this upcoming release.

File: assets/src/edit-story/components/helpCenter/toggle/test/index.js
Comment (swissspidy) [reply assets/src/edit-story/components/helpCenter/toggle/test/index.js:20]: It should not block this PR. It's just a note to myself to think about this.

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js
Comment (swissspidy) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js]: Alright then. I'm still not very comfortable just skipping tests ignoring potential bugs, but if I know it will get addressed soon then I suppose it should be ok.

**Pull request merged**
